### ISSUE TITLE
SSH Proxy broken in VSC 1.63.1

### ISSUE THREAD

**Issue by dale-conklin** [microsoft/vscode#139534]
Issue Type: <b>Bug</b>

I have a complicated SSH setup using an SSH tunnel to a remote machine. Everything has worked perfectly with this setup up until the VSC version 1.63.1. VSC 1.62.3 and previous versions work without a problem. 

My SSH setup uses SecureCRT. I open the SecureCRT terminal and authenticate to a remote SSH server. SecureCRT is then configured with "Port Forwarding" that attaches a listener to my loopback address on a specific port. This localhost port is used to proxy my SSH connections through the SSH bastion host to the final SSH server where my code is located. I am using the Remote-SSH plugin to accomplish this connection, however the plugin is not the problem, because it works fine in VSC 1.62.3.

Here is my .config file for the SSH connection.
Host iebdev22-loop
  Hostname 127.0.0.2
  Port 2222
  User user1
  IdentityFile C:\Users\user1\\.ssh\private_key_file

VS Code version: Code 1.62.3 (ccbaa2d27e38e5afa3e5c21c1c7bef4657064247, 2021-11-17T08:11:14.551Z)
OS version: Windows_NT x64 10.0.22000
Restricted Mode: No
Remote OS version: Linux x64 3.10.0-1160.45.1.el7.x86_64

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz (8 x 1800)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.89GB (1.98GB free)|
|Process Argv|--crash-reporter-id 64039b5a-d3a8-4ff4-91b7-55d4b61b3d07|
|Screen Reader|no|
|VM|44%|

|Item|Value|
|---|---|
|Remote|SSH: iebdev22-loop|
|OS|Linux x64 3.10.0-1160.45.1.el7.x86_64|
|CPUs|Intel(R) Xeon(R) CPU E5-2650 v2 @ 2.60GHz (32 x 3114)|
|Memory (System)|125.67GB (15.99GB free)|
|VM|0%|
</details><details><summary>Extensions (9)</summary>

Extension|Author (truncated)|Version
---|---|---
jupyter-keymap|ms-|1.0.0
remote-ssh|ms-|0.66.1
remote-ssh-edit|ms-|0.70.0
python|ms-|2021.12.1559732655
vscode-pylance|ms-|2021.12.2
jupyter|ms-|2021.10.1101450599
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.4
autodocstring|njp|0.5.4

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythontb:30283811
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30384386
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
vscop804cf:30404767
vscop940:30404999
vsrem710:30416614
vscexrecpromp3t2:30407763

```

</details>

**Comment by tanhakabir** [comment]
@dale-conklin would you also be able to share the logs from the Remote-SSH channel in the Output tab with the failed connection?

**Comment by dale-conklin** [comment]
Below are the logs from the working version. I'll upgrade and send the logs from the failed version next.

`[13:41:39.138] Log Level: 2
[13:41:39.143] remote-ssh@0.66.1
[13:41:39.143] win32 x64
[13:41:39.162] SSH Resolver called for "ssh-remote+iebdev22-loop", attempt 1
[13:41:39.163] "remote.SSH.useLocalServer": false
[13:41:39.163] "remote.SSH.showLoginTerminal": false
[13:41:39.163] "remote.SSH.remotePlatform": {"192.168.1.151":"linux","iebdev22":"linux","iebdev22-loop":"linux"}
[13:41:39.163] "remote.SSH.path": undefined
[13:41:39.163] "remote.SSH.configFile": undefined
[13:41:39.163] "remote.SSH.useFlock": true
[13:41:39.164] "remote.SSH.lockfilesInTmp": false
[13:41:39.164] "remote.SSH.localServerDownload": auto
[13:41:39.164] "remote.SSH.remoteServerListenOnSocket": false
[13:41:39.164] "remote.SSH.showLoginTerminal": false
[13:41:39.164] "remote.SSH.defaultExtensions": []
[13:41:39.164] "remote.SSH.loglevel": 2
[13:41:39.164] SSH Resolver called for host: iebdev22-loop
[13:41:39.164] Setting up SSH remote "iebdev22-loop"
[13:41:39.187] Using commit id "ccbaa2d27e38e5afa3e5c21c1c7bef4657064247" and quality "stable" for server
[13:41:39.190] Install and start server if needed
[13:41:39.194] Checking ssh with "ssh -V"
[13:41:39.234] > OpenSSH_for_Windows_8.1p1, 
[13:41:39.235] > LibreSSL 3.
[13:41:39.235] > 0.2

[13:41:39.240] Running script with connection command: ssh -T -D 56313 "iebdev22-loop" bash
[13:41:39.242] Terminal shell path: C:\WINDOWS\System32\cmd.exe
[13:41:39.546] > ]0;C:\WINDOWS\System32\cmd.exe
[13:41:39.546] Got some output, clearing connection timeout
[13:41:39.928] > 4a83122c89d7: running
[13:41:39.966] > Acquiring lock on /home/user1/.vscode-server/bin/ccbaa2d27e38e5afa3e5c21c1c7bef4657064247/vscode-remote-lock.user1.ccbaa2d27e38e5afa3e5c21c1c7bef4657064247
[13:41:39.997] > Found existing installation at /home/user1/.vscode-server/bin/ccbaa2d27e38e5afa3e5c21c1c7bef4657064247...
[13:41:40.011] > Checking /home/user1/.vscode-server/.ccbaa2d27e38e5afa3e5c21c1c7bef4657064247.log and /home/user1/.vscode-server/.ccbaa2d27e38e5afa3e5c21c1c7bef4657064247.pid for a running server
> Looking for server with pid: 18504
[13:41:40.065] > Found running server...
[13:41:40.087] >  
> *
> * Reminder: You may only use this software with Visual Studio family products,
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
> *
>  
> Checking server status on port 35560 with wget
[13:41:40.094] > 4a83122c89d7: start
> SSH_AUTH_SOCK====
> DISPLAY====
> webUiAccessToken====
> listeningOn==35560==
> osReleaseId==centos==
> arch==x86_64==
> tmpDir==/run/user/13229==
> platform==linux==
> unpackResult====
> didLocalDownload==0==
> downloadTime====
> installTime====
> extInstallTime====
> serverStartTime====
> connectionToken==11aaaa11-a1aa-1111-a1a1-11aaaa11a11a==
> 4a83122c89d7: end
[13:41:40.094] Received install output: 
SSH_AUTH_SOCK====
DISPLAY====
webUiAccessToken====
listeningOn==35560==
osReleaseId==centos==
arch==x86_64==
tmpDir==/run/user/13229==
platform==linux==
unpackResult====
didLocalDownload==0==
downloadTime====
installTime====
extInstallTime====
serverStartTime====
connectionToken==11aaaa11-a1aa-1111-a1a1-11aaaa11a11a==

[13:41:40.095] Remote server is listening on 35560
[13:41:40.095] Parsed server configuration: {"serverConfiguration":{"remoteListeningOn":{"port":35560},"osReleaseId":"centos","arch":"x86_64","webUiAccessToken":"","sshAuthSock":"","display":"","tmpDir":"/run/user/13229","platform":"linux","connectionToken":"11aaaa11-a1aa-1111-a1a1-11aaaa11a11a"},"installUnpackCode":""}
[13:41:40.100] Starting forwarding server. localPort 56316 -> socksPort 56313 -> remotePort 35560
[13:41:40.103] Forwarding server listening on 56316
[13:41:40.103] Waiting for ssh tunnel to be ready
[13:41:40.105] [Forwarding server 56316] Got connection 0
[13:41:40.106] Tunneled 35560 to local port 56316
[13:41:40.106] Resolved "ssh-remote+iebdev22-loop" to "127.0.0.1:56316"
[13:41:40.107] > 
[13:41:40.117] ------




[13:41:40.139] [Forwarding server 56316] Got connection 1
[13:41:40.198] [Forwarding server 56316] Got connection 2
[13:45:11.527] [Forwarding server 56316] Got connection 3`

**Comment by dale-conklin** [comment]
What was broken in version 1.63.1 is now fixed in 1.63.2. I updated to version 1.63.2 and everything is working correctly.
